dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java

```diff
@@ -463,9 +545,27 @@ public void updateIndex(Context context, boolean force) {
      * @throws SearchServiceException occurs when something went wrong with querying the solr server
      */
     @Override
-    public void cleanIndex(boolean force) throws IOException,
-        SQLException, SearchServiceException {
+    public void cleanIndex(boolean force) throws IOException, SQLException, SearchServiceException {
+        if (force) {
+            try {
+                getSolr().deleteByQuery(
```

Comment:
Why not use "*:*" as the query ?

Reply:
it is safe in case the local installation has customized the search index in a way that we don't know so that it includes now things maybe managed and indexed by other classes/services.